x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/configuration_step/use_saved_search.ts

```diff
@@ -36,19 +36,32 @@ export function useSavedSearch() {
   const { currentSavedSearch, currentDataView, kibanaConfig } = mlContext;
 
   const getQueryData = () => {
-    let qry: estypes.QueryDslQueryContainer = {};
+    let qry: any = {};
     let qryString;
 
     if (currentSavedSearch !== null) {
-      const { query } = getQueryFromSavedSearchObject(currentSavedSearch);
+      const { query, filter } = getQueryFromSavedSearchObject(currentSavedSearch);
```

Comment:
Not related to the changes in this PR, but if you clone a job created off a saved search, why does it always go into the advanced JSON editor?

Reply:
Remembered that this is because the job config does not store a reference to the saved search saved object ID, so we go straight to the advanced editor with the query DSL.